Reproducible Failure in S3BlobStoreRepositoryTests.testMultipleSnapshotAndRollback

```
./gradlew :plugins:repository-s3:test --tests "org.elasticsearch.repositories.s3.S3BlobStoreRepositoryTests.testMultipleSnapshotAndRollback" -Dtests.seed=F901373721E2E900 -Dtests.security.manager=true -Dtests.locale=en-SX -Dtests.timezone=IET -Dcompiler.java=12 -Druntime.java=11
```

fails every time (current `master`) with the following:

```
  1> Caused by: java.io.IOException: Unable to upload object [indices/IqtnNK-PRzmvOdry1yZ_0Q/3/__DnIqPmsWRtOdfmhPzc36bQ] using a single upload
  1>    at org.elasticsearch.repositories.s3.S3BlobContainer.executeSingleUpload(S3BlobContainer.java:338) ~[main/:?]
  1>    at org.elasticsearch.repositories.s3.S3BlobContainer.lambda$writeBlob$1(S3BlobContainer.java:106) ~[main/:?]
  1>    at java.security.AccessController.doPrivileged(Native Method) ~[?:?]
  1>    at org.elasticsearch.repositories.s3.SocketAccess.doPrivilegedIOException(SocketAccess.java:48) ~[main/:?]
  1>    at org.elasticsearch.repositories.s3.S3BlobContainer.writeBlob(S3BlobContainer.java:104) ~[main/:?]
  1>    at org.elasticsearch.repositories.blobstore.BlobStoreRepository.snapshotFile(BlobStoreRepository.java:1284) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
  1>    at org.elasticsearch.repositories.blobstore.BlobStoreRepository$2.doRun(BlobStoreRepository.java:1056) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
  1>    ... 5 more
  1> Caused by: com.amazonaws.services.s3.model.AmazonS3Exception: Internal Server Error (Service: Amazon S3; Status Code: 500; Error Code: 500 Internal Server Error; Request ID: fb9a4c3c-2fc7-7033-6b91-483335f735c8; S3 Extended Request ID: null)
  1>    at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1712) ~[aws-java-sdk-core-1.11.562.jar:?]
  1>    at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1367) ~[aws-java-sdk-core-1.11.562.jar:?]
  1>    at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1113) ~[aws-java-sdk-core-1.11.562.jar:?]
  1>    at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:770) ~[aws-java-sdk-core-1.11.562.jar:?]
  1>    at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:744) ~[aws-java-sdk-core-1.11.562.jar:?]
  1>    at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:726) ~[aws-java-sdk-core-1.11.562.jar:?]
  1>    at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:686) ~[aws-java-sdk-core-1.11.562.jar:?]
  1>    at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:668) ~[aws-java-sdk-core-1.11.562.jar:?]
  1>    at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:532) ~[aws-java-sdk-core-1.11.562.jar:?]
  1>    at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:512) ~[aws-java-sdk-core-1.11.562.jar:?]
  1>    at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4920) ~[aws-java-sdk-s3-1.11.562.jar:?]
  1>    at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4866) ~[aws-java-sdk-s3-1.11.562.jar:?]
  1>    at com.amazonaws.services.s3.AmazonS3Client.access$300(AmazonS3Client.java:389) ~[aws-java-sdk-s3-1.11.562.jar:?]
  1>    at com.amazonaws.services.s3.AmazonS3Client$PutObjectStrategy.invokeServiceCall(AmazonS3Client.java:5800) ~[aws-java-sdk-s3-1.11.562.jar:?]
  1>    at com.amazonaws.services.s3.AmazonS3Client.uploadObject(AmazonS3Client.java:1789) ~[aws-java-sdk-s3-1.11.562.jar:?]
  1>    at com.amazonaws.services.s3.AmazonS3Client.putObject(AmazonS3Client.java:1749) ~[aws-java-sdk-s3-1.11.562.jar:?]
  1>    at org.elasticsearch.repositories.s3.S3BlobContainer.lambda$executeSingleUpload$17(S3BlobContainer.java:335) ~[main/:?]
  1>    at org.elasticsearch.repositories.s3.SocketAccess.lambda$doPrivilegedVoid$0(SocketAccess.java:57) ~[main/:?]
  1>    at java.security.AccessController.doPrivileged(Native Method) ~[?:?]
  1>    at org.elasticsearch.repositories.s3.SocketAccess.doPrivilegedVoid(SocketAccess.java:56) ~[main/:?]
  1>    at org.elasticsearch.repositories.s3.S3BlobContainer.executeSingleUpload(S3BlobContainer.java:334) ~[main/:?]
  1>    at org.elasticsearch.repositories.s3.S3BlobContainer.lambda$writeBlob$1(S3BlobContainer.java:106) ~[main/:?]
  1>    at java.security.AccessController.doPrivileged(Native Method) ~[?:?]
  1>    at org.elasticsearch.repositories.s3.SocketAccess.doPrivilegedIOException(SocketAccess.java:48) ~[main/:?]
  1>    at org.elasticsearch.repositories.s3.S3BlobContainer.writeBlob(S3BlobContainer.java:104) ~[main/:?]
  1>    at org.elasticsearch.repositories.blobstore.BlobStoreRepository.snapshotFile(BlobStoreRepository.java:1284) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
  1>    at org.elasticsearch.repositories.blobstore.BlobStoreRepository$2.doRun(BlobStoreRepository.java:1056) ~[elasticsearch-8.0.0-SNAPSHOT.jar:8.0.0-SNAPSHOT]
  1>    ... 5 more
```

--- a/plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java
+++ b/plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobStoreRepositoryTests.java
@@ -121,11 +121,11 @@ public class S3BlobStoreRepositoryTests extends ESBlobStoreRepositoryIntegTestCa
         final String endpoint = "http://" + InetAddresses.toUriString(address.getAddress()) + ":" + address.getPort();
 
         return Settings.builder()
-            .put(Settings.builder()
-                .put(S3ClientSettings.ENDPOINT_SETTING.getConcreteSettingForNamespace("test").getKey(), endpoint)
-                // Disable chunked encoding as it simplifies a lot the request parsing on the httpServer side
-                .put(S3ClientSettings.DISABLE_CHUNKED_ENCODING.getConcreteSettingForNamespace("test").getKey(), true)
-                .build())
+            .put(S3ClientSettings.ENDPOINT_SETTING.getConcreteSettingForNamespace("test").getKey(), endpoint)
+            // Disable chunked encoding as it simplifies a lot the request parsing on the httpServer side
+            .put(S3ClientSettings.DISABLE_CHUNKED_ENCODING.getConcreteSettingForNamespace("test").getKey(), true)
+            // Disable request throttling because some random values in tests might generate too many failures for the S3 client
+            .put(S3ClientSettings.USE_THROTTLE_RETRIES_SETTING.getConcreteSettingForNamespace("test").getKey(), false)
             .put(super.nodeSettings(nodeOrdinal))
             .setSecureSettings(secureSettings)
             .build();
@@ -283,19 +283,5 @@ public class S3BlobStoreRepositoryTests extends ESBlobStoreRepositoryIntegTestCa
             exchange.sendResponseHeaders(HttpStatus.SC_INTERNAL_SERVER_ERROR, -1);
             exchange.close();
         }
-    }
-
-    // override here to mute only for S3, please remove this overload when un-muting
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/46218")
-    @Override
-    public void testSnapshotAndRestore() throws Exception {
-        super.testSnapshotAndRestore();
-    }
-
-    // override here to mute only for S3, pleaseremove this overload when un-muting
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/46219")
-    @Override
-    public void testMultipleSnapshotAndRollback() throws Exception {
-        super.testMultipleSnapshotAndRollback();
     }
 }

